levels/level02.ts

```diff
@@ -0,0 +1,162 @@
+        game.log('You loot 1 ' + fine + ' ' + typ + '.');
+      }
+    }
+
+    town.events = [
+      {
+        name: 'Angered Gods',
+        weight: 10,
+        predicate: (game: Game) => {
+          return game.party.status.angeredGods.active;
+        },
+        action: (game: Game) => {
+          const r = rollDie(20);
+          if (r <= 2) {
+            game.log('The volcano rumbles in the distance, and spews out magma. One party member is hit by a lava bomb and dies');
+            game.killPartyMembers(1);
+          } else {
+            game.log('The volcano rumbles in the distance.');
+          }
+          loot(game);
+        },
+      },
+      {
+        name: 'Digging lava irrigation',
+        weight: 1,
+        action: (game: Game) => {
+          const wis = rollDie(20) + calcmod(game.party.wis, [[0, -1], [6, 0],[12, +1]]); // Wisdom tells you not to stand in the way of pyroclastic flow
+          const dex = rollDie(20) + calcmod(game.party.dex, [[0, -1], [6, 0],[12, +1],[16, +2]]); // Dexterity helps you get out of the way when you do stand in the way of pyroclastic flow
+          if ( wis <= 7) {
+            if ( dex <= 12 ) {
+              game.log('Some members of your party dig an irrigation ditch to divert some of the lava flow away from the town. One party member is overcome by pyroclastic flow and dies.'); 
+              game.killPartyMembers(1);
+            } else { 
+              game.log('Some members of your party dig an irrigation ditch to divert some of the lava flow away from the town. One party member is nearly overcome by pyroclastic flow, but manages to escape with their life'); 
+            }
+          } else if ( wis > 7 && wis <= 18 ) {
+            game.log('Some members of your party dig an irrigation ditch to divirt some of the lava flow away from the town.');
+          } else {
+            if ( dex <= 13 ) {
+              game.log('Some members of your party dig an irrigation ditch to divirt some of the lava flow away from the town.');
+            } else { 
+              game.log('Some members of your party dig an irrigation ditch to divirt some of the lava flow away from the town. One party member tries to outrun the pyroclastic flow, but is overcome and dies.');
+              game.killPartyMembers(1);
+            }
+          }    
+          loot(game);
+          maybeAngerGods(game);
+        },
+      }
+    ];
+
+    // Boss State:
+
+    boss.name = 'Magma Elemental';
+    boss.str = 14;
+    boss.dex = 15;
+    boss.con = 9;
+    boss.int = 8;
+    boss.wis = 8;
+    boss.cha = 7;
+    boss.weapon.physical = -1; // 1 blunt damage
+    boss.weapon.elemental = -1; // 1 fire damage
+    boss.armor.physical = -1; // 1 blunt armor
+    boss.armor.elemental = -1; // 1 fire armor
+
+    boss.events = [
+      {
+        name: 'Charging attack',
+        weight: 1,
+        predicate: (game: Game) => {
+          return !game.boss.state.flag1 && !game.boss.state.flag2;
```

Comment:
Here's something you could do instead of accessing the flags with the generic names, https://github.com/vorble/greatestparty/blob/4add73fc17c01eab9e08af622304fb183a37f59b/levels/level01.ts#L34-L37